src/Workspaces/Core/Portable/Utilities/TaskExtensions.cs

```diff
@@ -18,7 +18,7 @@ public static T WaitAndGetResult<T>(this Task<T> task, CancellationToken cancell
         {
 #if DEBUG
             var threadKind = ForegroundThreadDataInfo.CurrentForegroundThreadDataKind;
-            if (threadKind != ForegroundThreadDataKind.Wpf && threadKind != ForegroundThreadDataKind.StaUnitTest)
+            if (threadKind != ForegroundThreadDataKind.Unknown)
```

Comment:
this should probably be ==.  If you change to that and everything passes, then :+1: from me.